### ISSUE TITLE
Beta 8 not handling dependencies correctly when processing

### ISSUE THREAD

**Issue by sathomas** [bdkjones/CodeKit2#128]
Really more of a followup to #97, but didn't want to confuse that feature request with a potential bug. It looks like CK is processing files in alphabetical order instead of accounting for dependencies. In my project:
- `initial.scss` compiles to `_initial.css`
- `index.kit` includes (within `<style>` tags) `_initial.css`

Whenever CK refreshes, it compiles `index.kit` before `initial.scss` (thus my guess at alphabetical order). If something changes in `initial.scss` (or its dependencies), then those changes are not reflected in the new `index.html`.

(Of course, if #97 was implemented, then I wouldn't need to use the `_initial.css` intermediate file and possibly all would be well. ;^)

**Comment by sathomas** [comment]
In case there's a question, problem still exists in beta 7

**Comment by sathomas** [comment]
Just checked and the problem's still there in beta 8

**Comment by bdkjones** [comment]
Ok, there was indeed a bug here. It took a while to track down. I am assuming that your `index.kit` file is importing the CSS file like this (notice the lack of quotes around the file name):

```
<style> <!-- @import _initial.css --> </style>
```

The bug was that the import linker was not handling the missing quote marks correctly after my latest round of optimizations. That has been fixed for Beta 11 and, when you save `initial.scss`, your Kit file will then recompile to capture the changes made in the compiled CSS file.

In the meantime, adding quotes around the filename is a workaround.

**Comment by sathomas** [comment]
Nope. Here's the line from my `index.kit` verbatim:

```
        <style><!-- @import '../intermediate/_initial.css' --></style>
```

BTW, you have to disable libsass to see the problem. It's only the Ruby compiler that causes a problem. More suggestion of a timing problem/race condition

**Comment by bdkjones** [comment]
I need the entire project showing the issue. Please post a download link here.

**Comment by sathomas** [comment]
It's the same project https://www.dropbox.com/s/suhsd0ghb9ypgrg/atltrails.zip

**Comment by bdkjones** [comment]
I've tested this over here and CodeKit is working as expected.

At the bottom of `initial.scss` I added this:

```
#nav-inner { background-color: red !important; }
```

And saved the file. CodeKit compiled the Sass file, then noticed the change to the CSS file, which subsequently triggered a re-compile of `index.kit`. With the project open in my browser, I watched the navigation bar change colors as I varied the background color declaration above. Note that I did turn off the libsass compiler.

The notifications in the upper-right corner appear in the wrong order, but that's no indication of the order in which CodeKit processes the files.

In short: I'm not seeing the behavior you describe. What modifications are you making that are not reflected in the re-compiled index.kit file? I need an _exact_ set of steps to reproduce.
